spot_driver/test/src/conversions/test_robot_state.cpp

```diff
@@ -0,0 +1,844 @@
+  estop_state_first.set_state_description("some_text");
+  robot_state.mutable_estop_states()->Add(std::move(estop_state_first));
+
+  google::protobuf::Timestamp timestamp_second;
+  timestamp_second.set_seconds(12);
+  timestamp_second.set_nanos(0);
+  ::bosdyn::api::EStopState estop_state_second;
+  estop_state_second.mutable_timestamp()->CopyFrom(timestamp_second);
+  estop_state_second.set_name("second_estop_name");
+  estop_state_second.set_type(::bosdyn::api::EStopState_Type::EStopState_Type_TYPE_SOFTWARE);
+  estop_state_second.set_state(::bosdyn::api::EStopState_State::EStopState_State_STATE_NOT_ESTOPPED);
+  estop_state_second.set_state_description("other_text");
+  robot_state.mutable_estop_states()->Add(std::move(estop_state_second));
+
+  // GIVEN some nominal clock skew
+  google::protobuf::Duration clock_skew;
+  clock_skew.set_seconds(1);
+
+  // WHEN we create an EStopStatesArray ROS message from the RobotState
+  const auto out = getEstopStates(robot_state, clock_skew);
+
+  // THEN the output message contains both estop states
+  ASSERT_THAT(out.estop_states, SizeIs(2));
+
+  // THEN each output estop state contains equivalent data to its corresponding input
+  EXPECT_THAT(out.estop_states, EStopStatesContains("estop_name", spot_msgs::msg::EStopState::TYPE_HARDWARE,
+                                                    spot_msgs::msg::EStopState::STATE_ESTOPPED, "some_text",
+                                                    timestamp_first, clock_skew));
+  EXPECT_THAT(out.estop_states, EStopStatesContains("second_estop_name", spot_msgs::msg::EStopState::TYPE_SOFTWARE,
+                                                    spot_msgs::msg::EStopState::STATE_NOT_ESTOPPED, "other_text",
+                                                    timestamp_second, clock_skew));
+}
+
+TEST(RobotStateConversions, TestGetJointStates) {
+  // GIVEN a RobotState containing two unique joint states
+  ::bosdyn::api::RobotState robot_state;
+  google::protobuf::Timestamp timestamp;
+  timestamp.set_seconds(15);
+  timestamp.set_nanos(0);
+  robot_state.mutable_kinematic_state()->mutable_acquisition_timestamp()->CopyFrom(timestamp);
+  setJointState(robot_state.mutable_kinematic_state()->add_joint_states(), "fl.hx", 0.1, 0.2, 0.3, 0.4);
+  setJointState(robot_state.mutable_kinematic_state()->add_joint_states(), "arm0.wr0", 0.5, 0.6, 0.7, 0.8);
+
+  // GIVEN some nominal clock skew
+  google::protobuf::Duration clock_skew;
+  clock_skew.set_seconds(1);
+
+  // GIVEN a prefix
+  constexpr auto prefix = "my_prefix/";
+
+  // WHEN we create an JointState ROS message from the RobotState
+  const auto out = getJointStates(robot_state, clock_skew, prefix);
+
+  // THEN a message is created
+  ASSERT_THAT(out.has_value(), IsTrue());
+
+  // THEN each vector element of the joint state contains a number of elements matching the number of input joint states
+  ASSERT_THAT(out->name, SizeIs(2));
```

Comment:
Is it the case that if any of these are false, we don't want to check the other values?

Reply:
I guess so because of the later expectation checks?

Reply:
> Is it the case that if any of these are false, we don't want to check the other values?

Yeah, the way the test is written currently it's not meaningful to check specific indices if the expected index doesn't exist.